x-pack/plugins/remote_clusters/public/plugin.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { CoreSetup, Plugin, CoreStart } from 'kibana/public';
+import { init as initBreadcrumbs } from './application/services/breadcrumb';
+import { init as initDocumentation } from './application/services/documentation';
+import { init as initHttp } from './application/services/http';
+import { init as initUiMetric } from './application/services/ui_metric';
+import { init as initNotification } from './application/services/notification';
+import { init as initRedirect } from './application/services/redirect';
+import { Dependencies } from './types';
+
+export class RemoteClustersUIPlugin implements Plugin<void, void, Dependencies, any> {
+  setup(
+    { notifications: { toasts }, http, getStartServices }: CoreSetup,
+    { management, usageCollection }: Dependencies
+  ) {
+    const esSection = management.sections.getSection('elasticsearch');
+
+    esSection!.registerApp({
+      id: 'remote_clusters',
+      title: i18n.translate('xpack.remoteClusters.appTitle', {
+        defaultMessage: 'Remote Clusters',
+      }),
+      mount: async ({ element, setBreadcrumbs }) => {
+        const [core] = await getStartServices();
+        const {
+          i18n: { Context: i18nContext },
+          docLinks,
+          fatalErrors,
+        } = core;
+
+        // Initialize services
+        initBreadcrumbs(setBreadcrumbs);
```

Comment:
@jloleysens This is the type of convention I was trying to get to. In my apps I am migrating I am instantiating the service inside the plugin `setup()` cycle.

Reply:
:+1: I guess in this pattern we would also not want the stateful services to be accessed via import in downstream code (as it currently is), e.g.,https://github.com/elastic/kibana/blob/1a41c8481aed1a48fc067d1b41dc064aac51c6f7/x-pack/plugins/remote_clusters/public/application/services/breadcrumb.ts#L21

More asking ^ :)

Reply:
Correct, these are still be accessed via import. I was on the fence to whether it made sense to refactor as part of this migration PR. I think it might be better to open up a separate PR once this is merged and move to context if that is the pattern we want to follow.